contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -296,9 +389,18 @@ abstract contract PoSValidatorManager is
         }
 
         address owner = $._posValidatorInfo[validationID].owner;
+
+        address rewardRecipient = $._rewardRecipients[validationID];
+
+        if (rewardRecipient == address(0)) {
+            rewardRecipient = owner;
+        } else {
+            delete $._rewardRecipients[validationID];
```

Comment:
The logic for mapping cleanup looks good to me now but would it make sense to add checks to tests that they are deleted as expected?

Ideally we would also have a data structure on top and be able to verify that they are empty as well but that's definitely beyond the scope of this PR. (although it would enable for looping as well to end all delegations)

Reply:
Since these are mappings getters won't help right? I was thinking of just asserting that any key that was used during the test is `address(0)` at the end of it but not strictly necessary.